### DISCUSSION TITLE
Argon skinning features and documentation

### DISCUSSION THREAD

**Discussion by Araeos** [ppy/osu#27321] [Q&A] [answered]
I've searched the open and closed issues, discussions and web for lazer skinning of the new skins (argon/argon pro) and have come to the conclusion that many elements are coded into the new skins that remain changeable in the legacy skins.
I assume the line `"InstantiationInfo": "osu.Game.Skinning.ArgonSkin, osu.Game"` in the `skininfo.json` refers to the specific class that renders the corresponding ui/gameplay elements. It might be feasible to recompile lazer, but distribution of modifications to this skin is practically impossible beyond what the skin editor provides.
The elements I'm trying to change are the followpoint (I have some animations from my legacy skins but they do not load) and the bright afterglow/persistency of the hitcircles after they have been hit to reduce clutter.

1. Is my understanding correct? Is there some documentation of this on here or anywhere else on what elements can be modded in the new skins?
2. Assuming it is accurate, are there any plans to make these new skins more moddable or to allow some kind of plugin system to hot-plug external skin classes made by the community?

**Comment by bdach** [comment, marked as answer]
There is no documentation because the skinning format is quite ad-hoc and should not be seen as final. We may arbitrarily change it to match requirements.

I do not foresee there being support for "skin plugins".

**Reply by Araeos** [reply]
With the understanding that nothing is final or set in stone, are there any plans to make the new skins customizable to the degree they replace legacy skins?

**Reply by bdach** [reply]
> are there any plans to make the new skins customizable to the degree they replace legacy skins?

We haven't committed to anything but I would certainly hope so.
